neo4j/errors.py

```diff
@@ -168,53 +168,20 @@ def transaction(self):
             return None
 
 
-class ClientError(BoltFailure):
-    """
-    """
-
-    transient = False
-
-
-class NotALeader(ClientError):
-    """
-    """
-
-    code = "Neo.ClientError.Cluster.NotALeader"
-
-
-class ForbiddenOnReadOnlyDatabase(ClientError):
-    """
-    """
-
-    code = "Neo.ClientError.General.ForbiddenOnReadOnlyDatabase"
-
-
-class DatabaseError(BoltFailure):
-    """
-    """
-
-    transient = False
-
-
-class TransientError(BoltFailure):
-    """
-    """
-
-    transient = True
-
-
-class Neo4jError(Exception):
-    """ Base class for all Neo4j service exceptions.
-    """
-
-
-class Neo4jAvailabilityError(Neo4jError):
+class BoltNeo4jAvailabilityError(BoltError):
```

Comment:
This is a very clumsy name. It should have either one prefix or the other, not both. How come this has moved from surface (`Neo4jError`) to internal (`BoltError`)?

Reply:
It was only used internally, to then surface a `ServiceUnavailable` error.

Reply:
Removed `BoltNeo4jAvailabilityError`  and replaced the ones with sub classes for `ServiceUnavailable`

+ ServiceUnavailable
     + RoutingServiceUnavailable
     + WriteServiceUnavailable
     + ReadServiceUnavailable

Commit: https://github.com/neo4j/neo4j-python-driver/pull/350/commits/06279ad6946e23741a680f6c300b119f5bcf997f